spec.html

```diff
@@ -21534,47 +22123,64 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
+            1. Return Completion(_element_).
+          1. Set _element_ to _element_.[[Value]].
+          1. If _element_.[[Kind]] is not ~field~ and _element_.[[Key]] is a Private Name, then
             1. If IsStatic of _e_ is *false*, let _container_ be _instancePrivateMethods_.
             1. Else, let _container_ be _staticPrivateMethods_.
-            1. If _container_ contains a PrivateElement whose [[Key]] is _field_.[[Key]], then
+            1. If _container_ contains a PrivateElement whose [[Key]] is _element_.[[Key]], then
               1. Let _existing_ be that PrivateElement.
-              1. Assert: _field_.[[Kind]] and _existing_.[[Kind]] are both ~accessor~.
-              1. If _field_.[[Get]] is *undefined*, then
-                1. Let _combined_ be PrivateElement { [[Key]]: _field_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _existing_.[[Get]], [[Set]]: _field_.[[Set]] }.
+              1. Assert: _element_.[[Kind]] is ~getter~ or ~setter~.
+              1. Assert: _existing_.[[Kind]] is ~accessor~.
+              1. If _element_.[[Get]] is *undefined*, then
+                1. Let _combined_ be PrivateElement { [[Key]]: _existing_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _existing_.[[Get]], [[Set]]: _element_.[[Set]] }.
               1. Else,
-                1. Let _combined_ be PrivateElement { [[Key]]: _field_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _field_.[[Get]], [[Set]]: _existing_.[[Set]] }.
+                1. Let _combined_ be PrivateElement { [[Key]]: _existing_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _element_.[[Get]], [[Set]]: _existing_.[[Set]] }.
               1. Replace _existing_ in _container_ with _combined_.
             1. Else,
-              1. Append _field_ to _container_.
-          1. Else if _field_ is a ClassFieldDefinition Record, then
-            1. If IsStatic of _e_ is *false*, append _field_ to _instanceFields_.
-            1. Else, append _field_ to _staticFields_.
+              1. If _element_.[[Kind]] is ~getter~ or ~setter~, let _kind_ be ~accessor~.
+              1. Else, let _kind_ be _element_.[[Kind]].
+              1. Let _privateElement_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: _kind_, [[Value]]: _element_.[[Value]], [[Get]]: _element_.[[Get]], [[Set]]: _element_.[[Set]] }.
+              1. Append _privateElement_ to _container_.
+          1. If IsStatic of _e_ is *false*, append _element_ to _instanceElements_.
+          1. Else, append _element_ to _staticElements_.
+        1. Let _parentInstanceMetadata_ be ? Get(_protoParent_, @@metadata).
+        1. Perform ! DefinePropertyOrThrow(_proto_, @@metadata, ConvertDecoratorMetadataListToObject(_instanceMetadataList_, _parentInstanceMetadata_)).
         1. Set the running execution context's LexicalEnvironment to _env_.
         1. If _classBinding_ is not *undefined*, then
           1. Perform _classScope_.InitializeBinding(_classBinding_, _F_).
```

Comment:
This should be set after class decorators run ([below](https://github.com/tc39/ecma262/pull/2417/files#diff-181371b08d71216599b0acccbaabd03c306da6de142ea6275c2135810999805aR22169-R22173)) to ensure this binding uses any replacement class returned by class decorators.

Reply:
This should be addressed now!